CMakeLists.txt

```diff
@@ -1,12 +1,21 @@
 cmake_minimum_required(VERSION 3.14.7)
 
+if(BOARD)
+    set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/cmake/stm32-gcc-arm-none-eabi.cmake)
```

Comment:
This is needed if not using a top level CMakeLists.txt so the compiler will still be fully defined